forschungsprojekte-II-ws2223/moodle-mod_jupyter
# Backup/Restore Fix

## Pull Request Template

### New features to test:
- Backing up and restoring the plugin should now work correctly: The uploaded file will be restored.
- A backup can be generated as follows:
  1. Create a course that cotains the plugin.
  2. Go to the main page of the course.
  3. Click on **"More"** (the rightmost option in the bar right below the course title), then Select **"Course reuse"** (the lowermost option).
  4. Select **"Backup"** in the dropdown menu on the left and create a backup (you can click on **"Jump to final step"** since the options do not matter).
  5. After the backup was created, you will be redirected to the **"Restore"** site; here you can just click **"Restore"** on the backup you just created.
  6. Finish the restoration process (i.e. by merging it into an existing course and deleting its previous contents) and **check whether the file you previously uploaded to the plugin is still there**.

#### Things to check before merging branches:

- [ ] The code was reviewed.
- [ ] The code conventions were complied with.
- [ ] The code complies with the naming conventions.
- [ ] The code is commented sufficiently.
- [ ] Issues related to the branch are closed.
- [ ] The feature related to the branch was tested if possible.
- [ ] The feature related to the branch is working.
- [ ] The pipeline has finished successfully.
- [ ] Unit tests were created and passed.
- [ ] Regression tests were passed.
- [ ] The documentation was updated if necessary.
